pkg/trtl/server.go

```diff
@@ -0,0 +1,156 @@
+package trtl
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/rotationalio/honu"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/trisacrypto/directory/pkg/trtl/config"
+	"github.com/trisacrypto/directory/pkg/trtl/pb/v1"
+	"github.com/trisacrypto/directory/pkg/trtl/peers/v1"
+	"github.com/trisacrypto/directory/pkg/utils/logger"
+	"google.golang.org/grpc"
+)
+
+func init() {
+	// Initialize zerolog with GCP logging requirements
+	zerolog.TimeFieldFormat = time.RFC3339
+	zerolog.TimestampFieldName = logger.GCPFieldKeyTime
+	zerolog.MessageFieldName = logger.GCPFieldKeyMsg
+
+	// Add the severity hook for GCP logging
+	var gcpHook logger.SeverityHook
+	log.Logger = zerolog.New(os.Stdout).Hook(gcpHook).With().Timestamp().Logger()
+}
+
+// A Trtl server implements the following services
+// 1. A database service for interacting with a database
+// 2. A peers management service for interacting with remote peers
+// 3. A replication service which implements auto-adapting anti-entropy replication.
+type Server struct {
+	srv     *grpc.Server    // The gRPC server that listens on its own independent port
+	conf    config.Config   // Configuration for the trtl server
+	db      *honu.DB        // Database connection for managing objects
+	trtl    *TrtlService    // Service for interacting with a Honu database
+	peers   *PeerService    // Service for managing remote peers
+	replica *ReplicaService // Service that handles anti-entropy replication
+	echan   chan error      // Channel for receiving errors from the gRPC server
+}
+
+// New creates a new trtl server given a configuration.
+func New(conf config.Config) (s *Server, err error) {
+	// Load the default configuration from the environment
+	if conf.IsZero() {
+		if conf, err = config.New(); err != nil {
+			return nil, err
+		}
+	}
+
+	// Set the global level
+	zerolog.SetGlobalLevel(conf.GetLogLevel())
+
+	// Set human readable logging if specified
+	if conf.ConsoleLog {
+		log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
+	}
+
+	// Create the server and prepare to serve
+	s = &Server{
+		conf:  conf,
+		echan: make(chan error, 1),
+		srv:   grpc.NewServer(grpc.UnaryInterceptor(s.interceptor)),
+	}
+
+	// TODO: check for maintenance mode
```

Comment:
Just a heads up that you'll need to handle a merge conflict issue here; we merged in the maintenance mode PR before you did the file renaming. I believe that merge will also impact the file now called `trtl/server_test.go`